File: examples/sdk/python/2_env_routing.py
# /// script
# dependencies = ["envilder>=0.3.1"]
# ///

# Environment-based routing: pick a different map file per environment
import os

from envilder import Envilder

env = os.environ.get("APP_ENV", "production")

secrets = Envilder.load(env, {
    "development": "../../../envilder.json",
    "staging": "../../../envilder.json",
    "production": "../../../envilder.json",
    "test": None,  # no secrets loaded
})

for key in secrets:
    print(f"{key} = {os.environ.get(key)}")

File: examples/sdk/nodejs/2_env_routing.ts
// Environment-based routing: pick a different map file per environment
import { Envilder } from '@envilder/sdk';

const env = process.env.APP_ENV ?? 'production';

const secrets = await Envilder.load(env, {
  development: '../../../envilder.json',
  staging: '../../../envilder.json',
  production: '../../../envilder.json',
  test: null, // no secrets loaded
});

for (const [key, value] of secrets) {
  console.log(`${key} = ${value}`);
}
